src/components/FormInput/FormInput.styles.ts

```diff
@@ -0,0 +1,69 @@
+import { TextField } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import { SIZE_CONFIGS } from './FormInput.constants';
+
+import type { StyledFormInputProps } from './FormInput.types';
+
+export const StyledFormInput = styled(TextField, {
+  shouldForwardProp: (prop) => (typeof prop === 'string' ? !prop.startsWith('$') : true),
+})<StyledFormInputProps>(({ theme, $variant, $size, $error }) => {
+  const sizeConfig = $size ? SIZE_CONFIGS[$size] : SIZE_CONFIGS['medium'];
+
+  return {
+    width: '100%',
+    '& .MuiInputBase-root': {
+      height: theme.spacing(Number(sizeConfig.height.replace('rem', '')) * 2),
+      padding: theme.spacing(0),
+      '& .MuiInputBase-input': {
+        padding: theme.spacing(
+          Number(sizeConfig.padding.split(' ')[0].replace('rem', '')) * 2,
+          Number(sizeConfig.padding.split(' ')[1].replace('rem', '')) * 2
+        ),
+        fontSize: sizeConfig.fontSize,
+        '&::placeholder': {
+          color: $error ? theme.palette.error.main : theme.palette.divider,
+          opacity: 1,
+        },
+      },
+    },
+    '& .MuiOutlinedInput-root': {
+      '& fieldset': {
+        borderColor: $error ? theme.palette.error.main : theme.palette.divider,
+        borderRadius: theme.shape.borderRadius,
+        borderWidth: '1px',
+      },
+      '&:hover fieldset': {
+        borderColor: $error ? theme.palette.error.main : theme.palette.custom?.focus,
```

Comment:
The style references theme.palette.custom?.focus. 'palette.custom' is not a standard MUI palette key and may be undefined at runtime or break theme typings unless the theme augmentation is present. Optional chaining prevents runtime crash but it hides the fact that the design token may not exist.

**Sugerencia:**
```typescript
Use a palette key that exists on your theme (e.g., primary.main, secondary.main) or ensure theme augmentation types define palette.custom.focus. If you rely on a custom token, add a fallback value, e.g.:

borderColor: $error ? theme.palette.error.main : (theme.palette.custom?.focus ?? theme.palette.primary.main),

and make sure to update theme types accordingly.
```